pyanaconda/modules/storage/partitioning/base.py

```diff
@@ -76,3 +79,72 @@ def validate_with_task(self):
         :return: a DBus path to a task
         """
         pass
+
+    def setup_kickstart(self, data):
+        """Setup the kickstart data."""
+        if not self._storage_playground:
+            return
+
+        self._setup_kickstart_from_storage(data, self._storage_playground)
+
+    @staticmethod
+    def _setup_kickstart_from_storage(data, storage):
+        """Setup the kickstart data from the given storage.
+
+        :param data: an instance of kickstart data
+        :param storage: an instance of the storage model
+        """
+        # Map devices on kickstart commands and data.
+        ks_map = {
+            PartitionDevice: ("PartData", "partition"),
+            TmpFSDevice: ("PartData", "partition"),
+            LVMLogicalVolumeDevice: ("LogVolData", "logvol"),
+            LVMVolumeGroupDevice: ("VolGroupData", "volgroup"),
+            MDRaidArrayDevice: ("RaidData", "raid"),
+            BTRFSDevice: ("BTRFSData", "btrfs")
+        }
+
+        # List comprehension that builds device ancestors should not get None
+        # as a member when searching for bootloader devices
+        bootloader_devices = []
+
+        if storage.bootloader_device is not None:
+            bootloader_devices.append(storage.bootloader_device)
+
+        for device in storage.devices:
+            if device.format.type == 'biosboot':
+                bootloader_devices.append(device)
+
+        # Make a list of ancestors of all used devices
+        used_devices = list(storage.mountpoints.values()) + storage.swaps + bootloader_devices
+        all_devices = list(set(a for d in used_devices for a in d.ancestors))
+        all_devices.sort(key=lambda d: len(d.ancestors))
```

Comment:
This is a nice candidate for a separate method.

Reply:
It is not my code and I don't want to mess with it.